src/main/java/frc/robot/subsystems/chassis/Drivetrain.java

```diff
@@ -0,0 +1,205 @@
+
+		// closed loop inversion configuration
+		/*frontLeft.setInverted(driveInverted);
+		frontRight.setInverted(!driveInverted);
+		backLeft.setInverted(driveInverted);
+		backRight.setInverted(!driveInverted);*/
+
+		frontLeft.setOpenLoopRampRate(Constants.Drivetrain.OPEN_LOOP_RAMP);
+		frontRight.setOpenLoopRampRate(Constants.Drivetrain.OPEN_LOOP_RAMP);
+		backLeft.setOpenLoopRampRate(Constants.Drivetrain.OPEN_LOOP_RAMP);
+		backRight.setOpenLoopRampRate(Constants.Drivetrain.OPEN_LOOP_RAMP);
+
+		frontLeft.setSmartCurrentLimit(Constants.Drivetrain.CURRENT_LIMIT);
+		frontRight.setSmartCurrentLimit(Constants.Drivetrain.CURRENT_LIMIT);
+		backLeft.setSmartCurrentLimit(Constants.Drivetrain.CURRENT_LIMIT);
+		backRight.setSmartCurrentLimit(Constants.Drivetrain.CURRENT_LIMIT);
+
+	}
+
+	// open loop curve drive method
+	public void curveDrive(double linearVelocity, double angularVelocity, boolean isQuickTurn) {
+		if (slowMode) {
+			differentialDrive.curvatureDrive(linearVelocity * Constants.Drivetrain.SLOW_SPEED, angularVelocity, isQuickTurn);
+		} else {
+			differentialDrive.curvatureDrive(linearVelocity, angularVelocity, isQuickTurn);
+		}
+	}
+
+	// closed loop drive method
+	public void closedCurveDrive(double linearVelocity, double angularVelocity, boolean isQuickTurn) {
+
+		ChassisSpeeds chassisSpeeds;
+		DifferentialDriveWheelSpeeds wheelSpeeds;
+		double leftOutput, rightOutput;
+		double leftFeedforwardOutput, rightFeedforwardOutput;
+
+		if (!isQuickTurn && linearVelocity > 0.02) {
+			chassisSpeeds = new ChassisSpeeds(linearVelocity * Constants.Drivetrain.CONTROLLER_LINEAR_SCALING, 0, angularVelocity * Constants.Drivetrain.CONTROLLER_ANGULAR_SCALING);
+			wheelSpeeds = driveKinematics.toWheelSpeeds(chassisSpeeds);
+
+			leftFeedforwardOutput = leftFeedforward.calculate(wheelSpeeds.leftMetersPerSecond);
+			rightFeedforwardOutput = rightFeedforward.calculate(wheelSpeeds.rightMetersPerSecond);
+
+			leftOutput = leftController.calculate(leftEncoder.getVelocity(), wheelSpeeds.leftMetersPerSecond);
+			rightOutput = rightController.calculate(rightEncoder.getVelocity(), wheelSpeeds.rightMetersPerSecond);
+
+			frontLeft.setVoltage(leftOutput + leftFeedforwardOutput);
+			frontRight.setVoltage(rightOutput + rightFeedforwardOutput);
+		} else {
+			chassisSpeeds = new ChassisSpeeds(0, 0, angularVelocity * Constants.Drivetrain.CONTROLLER_QUICKTURN_SCALING);
+			wheelSpeeds = driveKinematics.toWheelSpeeds(chassisSpeeds);
+
+			leftFeedforwardOutput = leftFeedforward.calculate(wheelSpeeds.leftMetersPerSecond);
+			rightFeedforwardOutput = rightFeedforward.calculate(wheelSpeeds.rightMetersPerSecond);
+
+			leftOutput = leftController.calculate(leftEncoder.getVelocity(), wheelSpeeds.leftMetersPerSecond);
+			rightOutput = rightController.calculate(rightEncoder.getVelocity(), wheelSpeeds.rightMetersPerSecond);
+
+			frontLeft.setVoltage(leftOutput + leftFeedforwardOutput);
+			frontRight.setVoltage(rightOutput + rightFeedforwardOutput);
+		}
```

Comment:
Most of the code here can be simplified since you repeat it.

```suggestion
		if (!isQuickTurn && linearVelocity > 0.02) {
			chassisSpeeds = new ChassisSpeeds(linearVelocity * Constants.Drivetrain.CONTROLLER_LINEAR_SCALING, 0, angularVelocity * Constants.Drivetrain.CONTROLLER_ANGULAR_SCALING);
		} else {
			chassisSpeeds = new ChassisSpeeds(0, 0, angularVelocity * Constants.Drivetrain.CONTROLLER_QUICKTURN_SCALING);
		}

		wheelSpeeds = driveKinematics.toWheelSpeeds(chassisSpeeds);

		leftFeedforwardOutput = leftFeedforward.calculate(wheelSpeeds.leftMetersPerSecond);
		rightFeedforwardOutput = rightFeedforward.calculate(wheelSpeeds.rightMetersPerSecond);

		leftOutput = leftController.calculate(leftEncoder.getVelocity(), wheelSpeeds.leftMetersPerSecond);
		rightOutput = rightController.calculate(rightEncoder.getVelocity(), wheelSpeeds.rightMetersPerSecond);

		frontLeft.setVoltage(leftOutput + leftFeedforwardOutput);
		frontRight.setVoltage(rightOutput + rightFeedforwardOutput);
```